doc/languages-frameworks/perl.md

```diff
@@ -0,0 +1,181 @@
+# Perl
+
+
+## Running perl programs on the shell
+
+When executing a Perl script, it is possible you get an error such as `./myscript.pl: bad interpreter: /usr/bin/perl: no such file or directory`. 
+This happens when the script expects Perl to be installed at `/usr/bin/perl`, which is not the case when using Perl from nixpkgs. 
+You can fix the script by changing the first line to:
+
+```
+#!/usr/bin/env perl
+```
+to take the Perl installation from the `PATH` environment variable, or invoke Perl directly with:
+
+```
+$ perl ./myscript.pl
+```
+
+When the script is using a Perl library that is not installed globally, you might get an error such as `Can't locate DB_File.pm in @INC 
+(you may need to install the DB_File module)`. In that case, you can use `nix-shell` to start an ad-hoc shell with that library installed, for instance:
+
+```
+$ nix-shell -p perl perlPackages.DBFile --run ./myscript.pl
+```
+
+If you are always using the script in places where `nix-shell` is available, you can embed the `nix-shell` invocation in the shebang like this:
+
+```
+#!/usr/bin/env nix-shell
+#! nix-shell -i perl -p perl perlPackages.DBFile
+```
+
+## Packaging Perl programs
+
+   Nixpkgs provides a function *buildPerlPackage*, a generic package builder function for any Perl package that has a standard *Makefile.PL*. It’s implemented in 
```

Comment:
```suggestion
   Nixpkgs provides a function `buildPerlPackage`, a generic package builder function for any Perl package that has a standard `Makefile.PL`. It’s implemented in 
```
I think it's more idiomatic in Markdown to use literals instead of emphasis for this kind of thing.

Reply:
Also the indentation looks weird.